frontend/public/components/utils/list-dropdown.jsx

```diff
@@ -175,22 +182,79 @@ ListDropdown.propTypes = {
   id: PropTypes.string,
 };
 
-const NsDropdown_ = (props) => {
-  const openshiftFlag = props.flags[FLAGS.OPENSHIFT];
-  if (flagPending(openshiftFlag)) {
-    return null;
+const useProjectOrNamespaceModel = () => {
+  const nsModel = modelFor('Namespace');
+  const canCreateNamespace = useAccessReview({
+    group: nsModel.apiGroup,
+    resource: nsModel.plural,
+    verb: 'create',
+  });
+
+  const canCreateProject = useFlag(FLAGS.CAN_CREATE_PROJECT);
+  const openshiftFlag = useFlag(FLAGS.OPENSHIFT);
+
+  if (flagPending(openshiftFlag) || flagPending(canCreateProject)) {
+    return [];
+  }
+
+  if (canCreateNamespace) {
+    return [modelFor('Namespace'), canCreateNamespace];
+  } else if (canCreateProject) {
+    return [modelFor('Project'), canCreateProject];
   }
-  const kind = openshiftFlag ? 'Project' : 'Namespace';
+};
+
+/** @type {React.FC<{dataFilter?: (ns: any) => boolean, desc?: string, selectedKey?: string, selectedKeyKind?: string, fixed?: boolean, placeholder?: string, onChange?: (selectedKey: string, event: React.Event) => void, id?: string, dataTest?: string}}>} */
+export const NsDropdown = (props) => {
+  const [selectedKey, setSelectedKey] = React.useState('');
+  const [model, canCreate] = useProjectOrNamespaceModel();
+
+  const actionItems =
+    model && canCreate
+      ? [
+          {
+            actionTitle: `Create ${model.label}`,
+            actionKey: `Create_${model.label}`,
+          },
+        ]
+      : [];
+
+  const onChange = (actionKey) => {
+    switch (actionKey) {
+      case 'Create_Namespace':
+        createNamespaceModal({
+          blocking: true,
+          onSubmit: (newNamespace) => {
+            setSelectedKey(newNamespace);
+          },
+        });
+        break;
+      case 'Create_Project':
+        createProjectModal({
+          blocking: true,
+          onSubmit: (newProject) => {
+            setSelectedKey(newProject);
+          },
+        });
+        break;
+      default:
+        break;
+    }
+  };
+
+  const kind = model.label;
   const resources = [{ kind }];
```

Comment:
If `model` is undefined here, which it can be, `model.label` will throw a runtime error. Even if we use optional chaining to prevent that, these variables will never get used when model is undefined, because we return null in that case. Better to just reference the properties of `model` below as you need them.

```suggestion
```